app/controllers/application_controller.rb

```diff
@@ -1,2 +1,30 @@
 class ApplicationController < ActionController::Base
+        helper_method :current_user
+        helper_method :logged_in?
+
+        protect_from_forgery with: :exception
+        
+        def current_user
+                User.find_by(id: session[:user_id])
+        end
+
+        def logged_in?
+                !current_user.nil?
+        end
+
+        private
+
+        def require_login
+                unless logged_in?
+                   flash[:error] = "You must be logged in to access this section"
+                   redirect_to login_path
+                end
+        end
+
+        def correct_user
```

Comment:
correct and current can confuse people
`is_valid_user?` sounds better

Reply:
This should be at the extended controller level.
Like, check the correct user for comment at the comment controller level.
Think about it.